appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/impl/MetricsServiceImpl.java

```diff
@@ -251,22 +252,26 @@ private static void collectRegistry(String contextName, MetricRegistry registry,
             for (Entry<MetricID, Metric> entry : ((MetricRegistryImpl) registry).getMetrics(name).entrySet()) {
                 MetricID metricID = entry.getKey();
                 Metric metric = entry.getValue();
-                MonitoringDataCollector metricCollector = tagCollector(contextName, metricID, collector);
-                if (metric instanceof Counting) {
-                    metricCollector.collect(toName(metricID, "Count"), ((Counting) metric).getCount());
-                }
-                if (metric instanceof SimpleTimer) {
-                    metricCollector.collect(toName(metricID, "Duration"), ((SimpleTimer) metric).getElapsedTime().toMillis());
-                }
-                if (metric instanceof Timer) {
-                    metricCollector.collect(toName(metricID, "MaxDuration"), ((Timer) metric).getSnapshot().getMax());
-                }
-                if (metric instanceof Gauge) {
-                    Object value = ((Gauge<?>) metric).getValue();
-                    if (value instanceof Number) {
-                        metricCollector.collect(toName(metricID,
-                                getMetricUnitSuffix(registry.getMetadata(name).unit())), ((Number) value));
+                try {
+                    MonitoringDataCollector metricCollector = tagCollector(contextName, metricID, collector);
+                    if (metric instanceof Counting) {
+                        metricCollector.collect(toName(metricID, "Count"), ((Counting) metric).getCount());
+                    }
+                    if (metric instanceof SimpleTimer) {
+                        metricCollector.collect(toName(metricID, "Duration"), ((SimpleTimer) metric).getElapsedTime().toMillis());
+                    }
+                    if (metric instanceof Timer) {
+                        metricCollector.collect(toName(metricID, "MaxDuration"), ((Timer) metric).getSnapshot().getMax());
+                    }
+                    if (metric instanceof Gauge) {
+                        Object value = ((Gauge<?>) metric).getValue();
+                        if (value instanceof Number) {
+                            metricCollector.collect(toName(metricID,
+                                    getMetricUnitSuffix(registry.getMetadata(name).unit())), ((Number) value));
+                        }
                     }
+                } catch (Exception ex) {
+                    LOGGER.log(Level.SEVERE, "Failing retrieving metric: " + metricID, ex);;
```

Comment:
Logging the full exception at severe may be a bit overkill.
Knowledge Is Power, but this block gets hit a number of times so you'll get a fairly chunky stack trace being printed numerous times if the app hasn't been deployed yet, and during undeployment.

Would a truncated message rather than the full stack trace be sufficient?

```
  Failing retrieving metric: MetricID{name='requestcount.server', tags=[]}
java.lang.IllegalStateException: MBean instance not found: amx:name=clusterjsp/server,pp=/mon/server-mon[server],type=web-request-mon
        at fish.payara.microprofile.metrics.jmx.MBeanExpression.getNumberValue(MBeanExpression.java:142)
        at fish.payara.microprofile.metrics.jmx.MBeanGuageImpl.getValue(MBeanGuageImpl.java:60)
        at fish.payara.microprofile.metrics.jmx.MBeanGuageImpl.getValue(MBeanGuageImpl.java:50)
        at fish.payara.microprofile.metrics.impl.MetricsServiceImpl.collectRegistry(MetricsServiceImpl.java:267)
        at fish.payara.microprofile.metrics.impl.MetricsServiceImpl.collect(MetricsServiceImpl.java:241)
        at fish.payara.monitoring.internal.data.InMemorySeriesRepository.collectAll(InMemorySeriesRepository.java:232)
        at fish.payara.monitoring.internal.data.InMemorySeriesRepository.collectSourcesToMemory(InMemorySeriesRepository.java:196)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: javax.management.InstanceNotFoundException: amx:name=clusterjsp/server,pp=/mon/server-mon[server],type=web-request-mon
        at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getMBean(DefaultMBeanServerInterceptor.java:1095)
        at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getAttribute(DefaultMBeanServerInterceptor.java:643)
        at com.sun.jmx.mbeanserver.JmxMBeanServer.getAttribute(JmxMBeanServer.java:678)
        at com.sun.enterprise.v3.admin.DynamicInterceptor.getAttribute(DynamicInterceptor.java:325)
        at fish.payara.microprofile.metrics.jmx.MBeanExpression.getAttribute(MBeanExpression.java:118)
        at fish.payara.microprofile.metrics.jmx.MBeanExpression.getNumberValue(MBeanExpression.java:132)
        ... 13 more
|#]
```